### PR TITLE
[DOC] Add doc for usage of confusion matrix

### PR DESCRIPTION
**Pull request by XiudingCai**
## Motivation

Add docs for useful_tools\confusion_matrix.md.

## Modification

1. Added mmpretrain\docs\en\useful_tools\confusion_matrix.md
2. Added mmpretrain\docs\zh_CN\useful_tools\confusion_matrix.md
3. Modified index.rst

### CODE REVIEW

File: docs/zh_CN/useful_tools/confusion_matrix.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+**输出图片**：
+
+<div align=center><img src="https://user-images.githubusercontent.com/26739999/210298124-49ae00f7-c8fd-488a-a4da-58c285e9c1f1.png" style=" width: auto; height: 40%; "></div>
+
+## 基础用法
+
+```python
+>>> import torch
+>>> from mmcls.evaluation import ConfusionMatrix
```
Comment (Ezra-Yu) [inline docs/zh_CN/useful_tools/confusion_matrix.md:47]: The library name has now been changed to mmpretrain.

File: docs/en/useful_tools/confusion_matrix.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+- `--show`: If or not to show the matplotlib visualization result of the confusion matrix, the default is `False`.
+- `--show-path`: If `show` is True, the path where the results are saved is visualized.
+- `--include-values`: Whether to add values to the visualization results.
+- `--cmap`: The color map used for visualization results, `cmap`, which defaults to `viridis`.
+
+* `--cfg-options`: Modifications to the configuration file, refer to [Learn about Configs](../user_guides/config.md).
+
+**Examples of use**:
+
+```python
```
Comment (fangyixiao18) [inline docs/en/useful_tools/confusion_matrix.md:33]:
```suggestion
```shell
```

File: docs/zh_CN/useful_tools/confusion_matrix.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- `checkpoint`：权重路径。
+- `--show`：是否展示混淆矩阵的 matplotlib 可视化结果，默认不展示。
+- `--show-path`：如果 `show` 为 True，可视化结果的保存路径。
+- `--include-values`：是否在可视化结果上添加数值。
+- `--cmap`：可视化结果使用的颜色映射图，即 `cmap`，默认为 `viridis`。
+- `--cfg-options`：对配置文件的修改，参考[学习配置文件](../user_guides/config.md)。
+
+**使用示例**：
+
+```python
```
Comment (fangyixiao18) [inline docs/zh_CN/useful_tools/confusion_matrix.md:32]:
```suggestion
```shell
```

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Pull request merged**